src/ussd_pmtct_rapidpro.js

```diff
@@ -1,24 +1,393 @@
+        });
+
         self.states.add("state_start", function(name) {
+            self.im.user.answers = {};
+            return new MenuState(name, {
+                question: $([
+                    "Welcome to the Department of Health's MomConnect (MC).",
+                    "",
+                    "Is {{msisdn}} the cell number of the mother who wants to sign up to HIV-related messages?"
+                    ].join("\n")).context({msisdn: utils.readable_msisdn(self.im.user.addr, "27")}),
+                error:
+                    "Sorry we don't understand. Please enter the number next to the mother's " +
+                    "answer.",
+                choices: [
+                    new Choice("state_check_subscription", "Yes"),
+                    new Choice("state_enter_msisdn", "No")
+                ]
+            });
+        });
+
+        self.add("state_enter_msisdn", function(name) {
+            return new FreeText(name, {
+                question:
+                    "Please enter the cell number of the mother who would like to sign up to " +
+                    "receive HIV-related messages from MomConnect e.g. 0813547654.",
+                check: function(content) {
+                    if(!utils.is_valid_msisdn(content, "ZA")) {
+                        return (
+                            "Sorry, we don't understand that cell number. Please enter 10 digit " +
+                            "cell number that the mother would like to get HIV-related messages " +
+                            "on, e.g. 0813547654.");
+                    }
+                    if(utils.normalize_msisdn(content, "ZA") === "+27813547654") {
+                        return (
+                            "We're looking for the mother's information. Please avoid entering " +
+                            "the examples in the messages. Enter the mother's details."
+                        );
+                    }
+                },
+                next: "state_check_subscription"
+            });
+        });
+
+        self.add("state_check_subscription", function(name, opts) {
+            var msisdn = utils.normalize_msisdn(
+                _.get(self.im.user.answers, "state_enter_msisdn", self.im.user.addr), "ZA");
+
+            return self.rapidpro.get_contact({urn: "tel:" + msisdn})
+                .then(function(contact) {
+                    self.im.user.set_answer("contact", contact);
+                    // Set the language if we have it
+                    if(_.isString(_.get(contact, "language"))) {
+                        return self.im.user.set_lang(contact.language);
+                    }
+                }).then(function() {
+                    // Delegate to the correct state depending on group membership
+                    var contact = self.im.user.get_answer("contact");
+                    if(self.contact_in_group(contact, _.concat(self.im.config.public_group_ids, self.im.config.clinic_group_ids))){
```

Comment:
I think we only want clinic registrations to be able to register for PMTCT, there are no PMTCT messages for public registrations.

Reply:
thanks, I missed this error

Reply:
okay did not know this, will remove the public subscriptions